app/helpers/remote_execution_helper.rb

```diff
@@ -12,30 +12,34 @@ def job_invocation_chart(invocation)
     options = { :class => 'statistics-pie small', :expandable => true, :border => 0, :show_title => true }
 
     if (bulk_task = invocation.task)
-      failed_tasks = bulk_task.sub_tasks.select { |sub_task| task_failed? sub_task }
-      cancelled_tasks, failed_tasks = failed_tasks.partition { |task| task_cancelled? task }
-      success = bulk_task.output['success_count'] || 0
-      cancelled = cancelled_tasks.length
-      failed = failed_tasks.length
-      pending = (bulk_task.output['pending_count'] || bulk_task.sub_tasks.count)
+      unless invocation.queued?
+        total = invocation.targeting.resolved? ? invocation.targeting.hosts.count : 0
+        failed, not_failed = bulk_task.sub_tasks.partition { |sub_task| task_failed? sub_task }
```

Comment:
Also, I would suggest putting the data gathering into JobInvocation model and write test for this.